sdk/python/feast/infra/materialization/local_engine.py

```diff
@@ -0,0 +1,322 @@
+from datetime import datetime
+from typing import Callable, Dict, List, Literal, Optional, Sequence, Tuple, Union
+
+import dask.dataframe as dd
+import pandas as pd
+import pyarrow as pa
+from tqdm import tqdm
+
+from feast.batch_feature_view import BatchFeatureView
+from feast.entity import Entity
+from feast.feature_view import DUMMY_ENTITY_ID, FeatureView
+from feast.infra.offline_stores.offline_store import OfflineStore
+from feast.infra.online_stores.online_store import OnlineStore
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.repo_config import FeastConfigBaseModel, RepoConfig
+from feast.stream_feature_view import StreamFeatureView
+from feast.type_map import python_values_to_proto_values
+from feast.value_type import ValueType
+
+from .batch_materialization_engine import (
+    BatchMaterializationEngine,
+    MaterializationJob,
+    MaterializationTask,
+)
+
+DEFAULT_BATCH_SIZE = 10_000
+
+
+class LocalMaterializationEngineConfig(FeastConfigBaseModel):
+    """Batch Materialization Engine config for local in-process engine"""
+
+    type: Literal["local"] = "local"
+    """ Type selector"""
+
+
+class LocalMaterializationJob(MaterializationJob):
+    def __init__(self, job_id: str) -> None:
+        super().__init__()
+        self._job_id: str = job_id
```

Comment:
can we make it `self.job_id`?

Reply:
There's a method called `job_id()` so I don't want to stomp over it.